### PR TITLE
Problem: Not enough information is visible to users/staff when a VM reaches error/deploy_error/user_deploy_error

### PR DESCRIPTION
**Pull request by steve-gregory**
Solution: Add 'extra' JSONField for InstanceStatusHistory to record fault message and traceback. [Show the information in the GUI](https://github.com/cyverse/troposphere/pull/672) when the status is in an error state.

## Description

- Include 'extra' in API Serializer
- Remove 'total_hours' from serializer (Invalid value)
- add status_name, only fill out in deploy_error states
- Metadata values constrained to 255 chars on Openstack
- Clear any fault metadata when 'deploying' metadata has been included

## Checklist before merging Pull Requests
- [ ] [Review and merge Troposphere companion PR](https://github.com/cyverse/troposphere/pull/672)
- [x] Reviewed and approved by at least one other contributor.
- [ ] If necessary, include a snippet in CHANGELOG.md

### CODE REVIEW

**Comment by julianpistorius** [comment]
Looks good. 👍

File: core/models/instance.py [hunk 1]
```diff
@@ -203,6 +203,9 @@ def update_history(
             size,
             task=None,
             tmp_status=None,
+            fault=None,
+            deploy_fault_message=None,
+            deploy_fault_trace=None,
```
Comment (lenards) [inline core/models/instance.py]: 👍

File: core/models/instance.py [hunk 2]
```diff
@@ -241,10 +254,11 @@ def update_history(
             # logger.info("status_name matches last history:%s " %
             #        last_history.status.name)
             return (False, last_history)
+        status = self.esh_status()
```
Comment (lenards) [inline core/models/instance.py]: does it make sense to pull this line up to Line 225 with `activity = self.esh_activity()`?

File: core/models/instance.py [hunk 3]
```diff
@@ -220,15 +223,25 @@ def update_history(
...
+                start_date=self.start_date,
+                activity=activity,
+                extra=extra)
             last_history.save()
+            status = self.esh_status()
             logger.debug("STATUSUPDATE - FIRST - Instance:%s Old Status: %s - %s New\
                 Status: %s Tmp Status: %s" % (self.provider_alias,
-                                              self.esh_status(),
+                                              status,
                                               self.esh_activity(),
```
Comment (lenards) [inline core/models/instance.py]: I think this could be `activity` now with the local at Line 225.

**Comment by steve-gregory** [comment]
Self-:shipit:

**Comment by lenards** [comment]
Coveralls was deactivated, I thought.

**Pull request merged**